app/views/layouts/application.html.erb

```diff
@@ -1,14 +1,46 @@
 <!DOCTYPE html>
 <html>
-<head>
-  <title>GistList</title>
-  <%= stylesheet_link_tag    'application', media: 'all' %>
-  <%= javascript_include_tag 'application' %>
-  <%= csrf_meta_tags %>
-</head>
-<body>
+  <head>
+    <title>Gist-List</title>
+    <%= stylesheet_link_tag    'application', media: 'all' %>
+    <%= javascript_include_tag 'application' %>
+    <%= csrf_meta_tags %>
+  </head>
+  <body>
+    <% if flash.any?  %>
+      <div id="flash-container">
+        <% flash.each do |name, msg| %>
+          <%= content_tag :div, msg, class: "flash-#{name}" %>
+        <% end %>
+      </div>
+    <% end %>
+    <div id="header">
+          <h1 id="header-title"><a href="/">Welcome to Gist-List!</a></h1>
+          <nav>
+              <a href="/groups/new">Make A New Group</a> |
+              <a href="/groups">All Groups</a> |
+              <%= link_to 'All Snippets', snippets_path%>
 
-<%= yield %>
+              <% if !!session[:user_id] && current_user %>
```

Comment:
current_user already checks that there's an id in the session before it tries to load, so you don't need that additional check here.